File: frontend/src/api/system.ts
import client from './client'

export const systemApi = {
  health(): Promise<{ status: string }> {
    return client.get('/health').then((r) => r.data)
  },
  version(): Promise<{ name: string; version: string }> {
    return client.get('/version').then((r) => r.data)
  },
}

File: app/api/v1/system.py
"""System endpoints: health, version."""
from fastapi import APIRouter

from app import __version__

router = APIRouter(tags=["system"])


@router.get("/health")
def health() -> dict:
    return {"status": "ok"}


@router.get("/version")
def version() -> dict:
    return {"name": "pyquiz-forge", "version": __version__}
